### ISSUE TITLE
Rendering problem in PDF when depth of the nested 'ol' elements is greater than 4

### ISSUE THREAD

**Issue by GanZ2016** [dita-ot/dita-ot#3545]
## Expected Behavior
![image](https://user-images.githubusercontent.com/25555914/87212943-011af800-c2e7-11ea-932b-2822dde2aa14.png)

## Actual Behavior

![image](https://user-images.githubusercontent.com/25555914/87212889-94076280-c2e6-11ea-8771-bd9132db4242.png)

## Steps to Reproduce

1. Create a new DITA Topic
2.  Add the next `<ol/>` element in a DITA topic: 
```
<ol id="ol_pfg_ywn_hmb">
    <li>Level 1 - 1: <ol id="ol_f2s_ywn_hmb">
            <li>Level 2 - 1:<ol id="ol_gzb_swn_hmb">
                    <li>Level 3 - 1:<ol id="ol_hzb_swn_hmb">
                            <li>Level 4 - 1.</li>
                            <li>Level 4 - 2.</li>
                            <li>Level 4 - 3.</li>
                            <li>Level 4 - 4.<ol id="ol_izb_swn_hmb">
                                    <li>Level 5 - 1.</li>
                                    <li>Level 5 - 2.</li>
                                </ol></li>
                            <li>Level 4 - 5.</li>
                        </ol></li>
                </ol></li>
        </ol></li>
</ol>
```
3. Publish to PDF in oXygen using built-in "DITA PDF - based on XSL-FO".

## Copy of the error message, log file or stack trace

```
[DOTX052W]: No string named 'Ordered List Format 5' was found when creating generated text; using the value 'Ordered List Format 5' in your output file.
[DOTX052W]: No string named 'Ordered List Number 5' was found when creating generated text; using the value 'Ordered List Number 5' in your output file.
[DOTX052W]: No string named 'Ordered List Format 5' was found when creating generated text; using the value 'Ordered List Format 5' in your output file.
[DOTX052W]: No string named 'Ordered List Number 5' was found when creating generated text; using the value 'Ordered List Number 5' in your output file.
```

## Environment

* DITA-OT version: DITA-OT 3.x
* Operating system and version: Windows 10
  _(Linux, macOS, Windows)_
* How did you run DITA-OT? oXygen
  _(`dita` command, Ant, `startcmd`, oXygen, other editor, CMS, etc.)_
* Transformation type: PDF
  _(HTML5, PDF, custom, etc.)_

**Comment by raducoravu** [comment]
There is a configuration file

"DITA-OT/plugins/org.dita.pdf2/cfg/common/vars/commonvariables.xml"

which contains variables like:

      <variable id="Ordered List Number 4"><param ref-name="number"/>. </variable>

and: 

      <variable id="Ordered List Format 4">a</variable>

You probably need to override it in your PDF customization plugin and add more variables to it for your extra levels of nesting.

@jelovirt @infotexture what's your opinion on this, should we maybe have some documentation about this or maybe the XSLT stylesheets when they cannot find a certain level of list item should cycle around, for example for level 5 they should use the level 1 variables, for level 6 use the level 2 variables and so on?

**Comment by raducoravu** [comment]
One more thing, it's bad practice in general to have that many levels of lists in your content, it makes the content hard to follow and understand.

**Comment by infotexture** [comment]
I assume this is by design, as heavily nested lists are generally frowned upon by most style guides.

For example, the _IBM Style Guide_ provides this guidance:

> Avoid nesting ordered and unordered lists beyond two levels, and never nest more than three levels; 
> restructure the information instead.

**Comment by GanZ2016** [comment]
Thanks for the solution. @raducoravu  

We have a lot of heavily nested lists in our content, so that would be impossible to update them all.

I noticed in the `DITA-OT/plugins/org.dita.pdf2/xsl/fo/list.xsl`, you use `(($depth - 1) mod 4) + 1)` instead of `$depth` for "Unordered List bullet". 

I would suggest the same logic for "Ordered List Format" and "Ordered List Number" by replacing `$depth` with `(($depth - 1) mod 4) + 1)` to solve the problem.
